### ISSUE TITLE
Remove lock example code

### ISSUE THREAD

**Issue by chalin** [etcd-io/website#188]
There are `.go` source files and a README being published for a **lock example**, see:

- [lock example README][] -- I've inlined an excerpt below for your convenience
- https://etcd.io/docs/next/learning/lock/client/client.go
- https://etcd.io/docs/next/learning/lock/storage/storage.go

Questions for @spzala @ptabor:

- Like for the etcd mixin (https://github.com/etcd-io/etcd/pull/12686), should the **lock example source files** be moved to `etcd-io/etcd`'s **`contrib/` folder**?
- If so, should we:
  - Move the entire [etcd-io/website/content/docs/next/learning/lock][lock] directory, and hence, including the README, or
  - Keep the README and rewrite it a bit so that it reads as a proper Learning page on the website?

[lock example README]: https://etcd.io/docs/next/learning/lock/readme/
[lock]: https://github.com/etcd-io/website/tree/master/content/docs/next/learning/lock

Context:

- https://github.com/etcd-io/etcd/pull/11490

Steps:

- [x] Move lock example to `etcd/contrib` - https://github.com/etcd-io/etcd/pull/12836
- [x] Remove lock example sources from this repo - #191
- [x] Open an issue to track rework to the README, possibly creating a blog entry from it (although that might best happen after #94) --> #200

---

README excerpt from https://etcd.io/docs/next/learning/lock/readme/:

> ![image](https://user-images.githubusercontent.com/4140793/113734837-af629b00-96c9-11eb-9cbb-b71a0d4f94b9.png)

**Comment by nate-double-u** [comment]
I'd like to see this moved back to the `etcd-io/etcd` `contrib/` folder, but I'd also like to see the content reworked to better fit the docs site. 

Maybe This is a good blog entry?

**Comment by chalin** [comment]
Ok, how about doing this in two steps: (1) relocate, (2) create a followup issue to rework the content. That way we'll be able to continue making progress (w/o extra delay) towards our pre-docsy cleanup?

**Comment by nate-double-u** [comment]
> Open an issue to track rework to the README, possibly creating a blog entry from it (although that might best happen after #94)

Just a note, I closed #94 in favour of https://github.com/etcd-io/website/issues/134

**Comment by chalin** [comment]
The task is covered by #200. Closing this issue since the example has been removed.
